### DISCUSSION TITLE
ender 3 v3

### DISCUSSION THREAD

**Discussion by bhast2** [Guilouz/Creality-Helper-Script-Wiki#472] [Features Request]
Is there any to use this script on the new Creality Ender 3 v3 it is pretty much a k1 but corexz but it is using the say screen and Creality OS. thank you

**Comment by TheWebMachine** [comment]
Highly doubtful. In part because it is coreXZ, many of the custom macros would need to be quite different. 

There's nothing stopping someone from forking this project and creating an E3V3 version with properly customized scripts, however. 🤓

**Comment by Guilouz** [comment]
My script is compatible with KE Series. All K1 features not compatible with KE are not displayed in script.

**Reply by TheWebMachine** [reply]
There's the KE here:
https://www.creality.com/products/creality-ender-3-v3-ke

But I think they are referring to the slightly newer V3 (no additional suffix), which is basically a K1/K1C without the camera and enclosure:
https://www.creality.com/products/creality-ender-3-v3

I suppose your script would only need some minor rework, then. I forgot you support KE, another CoreXZ. 🤦🏻‍♂️

**Comment by bhast2** [comment]
Yes the printer I am talking about is the ender 3 v3. Not the KE or the SE

**Comment by bhast2** [comment]
So will it work for the new Ender 3 V3 or does something need to be done to it to work.  Here is a video on it
https://www.youtube.com/watch?v=AyAuHlcz4QM

**Comment by doubletaco** [comment]
I wanted to install Mainsail since I prefer that UI, so I ran this script against an up-to-date Ender 3 V3. Here's my experience.
1. The script runs but does not detect existing running Fluid, Moonraker, or Nginx.
2. I had to run the Install Moonraker/Nginx script before it would let me install Mainsail.
3. This broke the existing Moonraker and Fluid installations. I re-ran the script and removed Moonraker, Nginx, and Mainsail.
4. This left the printer's Web UI unusable. The existing Fluid and Moonraker had either been removed, or broken as part of the first installation.
5. I re-ran the installation script for Moonraker/Nginx. This time I got the Moonraker screen back on port 7125.
6. I reinstalled Fluid and Mainsail using the helper script. Both now work on their respective ports, 4408 and 4409.
7. I installed Mobileraker Companion for push notifications and that also seems to be working.
Because it detects the machine as a KE (my guess is this is the fallback if it's not a K1/Max) there are no options to install things like KAMP.
As of now, with limited testing, the printer has not exploded and both web front ends are working and displaying accurate print details. I have no idea what updating firmware will do later down the line, but I'm happy to have my Mainsail experience ready to go.

**Comment by bhast2** [comment]
A possible way to add this machine to the script so it doesn't think it is a KE? And sounds like a bit of a chance of messing up your printer by doing what @doubletaco did. I am getting my printer today and don't want to brick it lol

**Reply by doubletaco** [reply]
For what it's worth, my printer is working fine. Also nothing stopping us from manually installing KAMP. I did that too following the instructions in their github page: https://github.com/kyleisah/Klipper-Adaptive-Meshing-Purging 
Just remember to add the relevant features to your START_PRINT macro in gcode_macros.cfg

**Reply by bhast2** [reply]
mind sharing you sharing your start_print macros before I take the jump?  Thank you

**Comment by bhast2** [comment]
@Guilouz if there is anything you need from me to help you to add this printer to your wonderful script let me know.

**Comment by Guilouz** [comment]
It’s difficult to add support when i don’t have this machine. I have asked to Creality to send me one but not sure.

**Reply by bhast2** [reply]
Yeah I wasn't sure if I could supply the printer.cfg and other cfg files if that would help you.

**Comment by bhast2** [comment]
So I got your script to work somewhat. Moonraker and fluid installed and do did timelapse and they work just fine. I made sure I had backups of my cfg files before I started with the script. I have printed two items and it is working really well.

**Comment by bhast2** [comment]
So the ender 3v3 has the same start code as a k1 and k1 max. Please advise on how to add kamp to this. Thank you. 

```
[gcode_macro START_PRINT]
variable_prepare: 0
variable_z_sensorless: 0
gcode:
  SET_GCODE_VARIABLE MACRO=M106 VARIABLE=user_flag VALUE=0
  WAIT_TEMP_END
  CLEAR_PAUSE
  SET_TEMPERATURE_FAN_TARGET temperature_fan=soc_fan target=5
  {% set g28_extruder_temp = printer.custom_macro.g28_ext_temp %}
  {% set bed_temp = printer.custom_macro.default_bed_temp %}
  {% set extruder_temp = printer.custom_macro.default_extruder_temp %}

  {% if 'BED_TEMP' in params|upper and (params.BED_TEMP|float) %}
    {% set bed_temp = params.BED_TEMP %}
  {% endif %}

  {% if 'EXTRUDER_TEMP' in params|upper and (params.EXTRUDER_TEMP|float) %}
    {% set extruder_temp = params.EXTRUDER_TEMP %}
  {% endif %}

  {% if printer['gcode_macro START_PRINT'].prepare|int == 0 %}
    {action_respond_info("not prepare.\n")}
    
    PRINT_PREPARE_CLEAR
    CX_ROUGH_G28 EXTRUDER_TEMP={extruder_temp} BED_TEMP={bed_temp}
    CX_NOZZLE_CLEAR
    Z_SENSORLESS_SET
    ACCURATE_G28
  {% else %}
    PRINT_PREPARE_CLEAR
  {% endif %}

  CX_PRINT_DRAW_ONE_LINE
```

**Reply by doubletaco** [reply]
Assuming all your printer.cfg and KAMP_Settings.cfg stuff is set up, here are the changes I made. I'm using adaptive meshing, smart park, and line purging.

after ACCURATE_G28 add
```
BED_MESH_CLEAR
BED_MESH_CALIBRATE
SMART_PARK
```

then comment out `CX_PRINT_DRAW_ONE_LINE` and add `LINE_PURGE` under the commented out line.

**Reply by bhast2** [reply]
I didn't add SMART_PARK yet. but i was getting a temp error before the LINE_PURGE so here is my start_print macro, and it works great!!!!

```
[gcode_macro START_PRINT]
variable_prepare: 0
variable_z_sensorless: 0
gcode:
  SET_GCODE_VARIABLE MACRO=M106 VARIABLE=user_flag VALUE=0
  WAIT_TEMP_END
  CLEAR_PAUSE
  SET_TEMPERATURE_FAN_TARGET temperature_fan=soc_fan target=5
  {% set g28_extruder_temp = printer.custom_macro.g28_ext_temp %}
  {% set bed_temp = printer.custom_macro.default_bed_temp %}
  {% set extruder_temp = printer.custom_macro.default_extruder_temp %}

  {% if 'BED_TEMP' in params|upper and (params.BED_TEMP|float) %}
    {% set bed_temp = params.BED_TEMP %}
  {% endif %}

  {% if 'EXTRUDER_TEMP' in params|upper and (params.EXTRUDER_TEMP|float) %}
    {% set extruder_temp = params.EXTRUDER_TEMP %}
  {% endif %}

  {% if printer['gcode_macro START_PRINT'].prepare|int == 0 %}
    {action_respond_info("not prepare.\n")}
    
    PRINT_PREPARE_CLEAR
    CX_ROUGH_G28 EXTRUDER_TEMP={extruder_temp} BED_TEMP={bed_temp}
    CX_NOZZLE_CLEAR
    Z_SENSORLESS_SET
    ACCURATE_G28
    BED_MESH_CLEAR
    BED_MESH_CALIBRATE
  {% else %}
    PRINT_PREPARE_CLEAR
  {% endif %}
    M109 S{extruder_temp}
    M190 S{bed_temp}
   LINE_PURGE
  #CX_PRINT_DRAW_ONE_LINE
```
